pkg/cli/cli_handle.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/abcxyz/access-on-demand/apis/v1alpha1"
+	"github.com/abcxyz/access-on-demand/pkg/handler"
+	"github.com/abcxyz/access-on-demand/pkg/requestutil"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/posener/complete/v2/predict"
+)
+
+var _ cli.Command = (*CLIHandleCommand)(nil)
+
+// CLIHandleCommand handles CLI requests.
+type CLIHandleCommand struct {
+	cli.BaseCommand
+
+	flagPath string
+
+	flagDebug bool
+
+	// Run Cleanup instead of Do if true.
+	Cleanup bool
+
+	// testCLI is used for testing only.
+	testCLI string
+}
+
+func (c *CLIHandleCommand) Desc() string {
+	return `Handle the CLI request YAML file at the given path`
+}
+
+func (c *CLIHandleCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+Run "do" commands in the CLI request YAML file at the given path:
+
+      aod cli do -path "/path/to/file.yaml"
+
+
+Run "do" commands in the CLI request YAML file at the given path in debug mode:
+
+      aod cli do -path "/path/to/file.yaml" -debug
+
+Run "cleanup" commands in the CLI request YAML file at the given path:
+
+      aod cli cleanup -path "/path/to/file.yaml"
+
+
+Run "cleanup" commands in the CLI request YAML file at the given path in debug mode:
+
+      aod cli cleanup -path "/path/to/file.yaml" -debug
+`
+}
+
+func (c *CLIHandleCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "path",
+		Target:  &c.flagPath,
+		Example: "/path/to/file.yaml",
+		Predict: predict.Files("*"),
+		Usage:   `The path of CLI request file, in YAML format.`,
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "debug",
+		Target:  &c.flagDebug,
+		Default: false,
+		Usage:   `Turn on debug mode to print command outputs.`,
+	})
+
+	return set
+}
+
+func (c *CLIHandleCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	if c.flagPath == "" {
+		return fmt.Errorf("path is required")
+	}
+
+	return c.do(ctx)
+}
+
+func (c *CLIHandleCommand) do(ctx context.Context) error {
+	// Read request from file path.
+	var req v1alpha1.CLIRequest
```

Comment:
Use pointer here?

Reply:
```var req *v1alpha1.CLIRequest``` will panic here ```panic: reflect: reflect.Value.Set using unaddressable value```
It needs to be either this ```var req v1alpha1.CLIRequest``` or ```req := &v1alpha1.CLIRequest{}```

Reply:
I think the code as-is is correct. We're passing around the pointer (`&req`)